src/python/gudhi/datasets/remote.py

```diff
@@ -204,20 +241,85 @@ def fetch_bunny(file_path = None, accept_license = False):
+_activities_license_checksum = "f5ce6749fa9d5359d7b0c4c37d0b61e5d9520f9494cd53be94295d3967ee4023"
+
+
+def fetch_daily_activities(file_path=None, subset="all", accept_license=False):
+    """
+    Load a subset of the Daily and Sports Activities dataset. This dataset comes from
+    https://archive.ics.uci.edu/ml/datasets/daily+and+sports+activities (CC BY 4.0 license).
+
+    Note that if the dataset already exists in the target location, it is not downloaded again,
+    and the corresponding dataset is read (compressed) from cache.
+
+    Parameters
+    ----------
+    file_path : string
+        Full path of the downloaded file including filename.
+
+        Default is None, meaning that it's set to "data_home/points/activities/activities_p1_left_leg.npy".
+        In this case, the LICENSE file would be downloaded as "data_home/points/activities/activities.LICENSE".
+
+        The "data_home" directory is set by default to "~/gudhi_data",
+        unless the 'GUDHI_DATA' environment variable is set.
+
+    subset : string
+        This argument allows to download the following subsets:
+         * 'all' (default value) This dataset contains 30.000 vertices in dimension 3 + activity type column ()
+         * 'cross_training' Only left leg magnetometer of cross training activity performed by the person 1. It contains 7.500 vertices in dimension 3.
+         * 'jumping' Only left leg magnetometer of jumping activity performed by the person 1. It contains 7.500 vertices in dimension 3.
+         * 'stepper' Only left leg magnetometer of stepper activity performed by the person 1. It contains 7.500 vertices in dimension 3.
+         * 'walking' Only left leg magnetometer of walking activity performed by the person 1. It contains 7.500 vertices in dimension 3.
+
+    accept_license : boolean
+        Flag to specify if user accepts the file LICENSE and prevents from printing the corresponding license terms.
+
+        Default is False.
+
+    Returns
+    -------
+    points: numpy array
+        Depending on subset value:
+         * Array of shape (30.000, 4, dtype = object) when `subset='all'`.
+
+        Or
+         * Array of shape (7.500, 3, dtype = double) otherwise.
+
+    """
+    file_url = (
+        "https://raw.githubusercontent.com/GUDHI/gudhi-data/main/points/activities/activities_p1_left_leg.npy"
+    )
+    file_checksum = "83a3b7cfc0cfce60ff249bebcf65c87bf5ab1dc45f2a13e40172b203d0f6285d"
+    gudhi_data_set_path = "points/activities/activities_p1_left_leg.npy"
+
+    archive_path = _get_archive_path(file_path, gudhi_data_set_path)
+    if not exists(archive_path):
+        _fetch_remote(file_url, archive_path, file_checksum)
+        license_path = join(split(archive_path)[0], "activities.LICENSE")
+        _fetch_remote_license(_activities_license_url, license_path, _activities_license_checksum, accept_license)
+
+    ds = np.load(archive_path, allow_pickle=True)
```

Comment:
The [data](https://github.com/GUDHI/gudhi-data/blob/main/points/activities/activities_p1_left_leg.npy) is a numpy array of type object, because I added the activity as a string in the last column.
With numpy array of type object are picklized (compressed), so I need to load it with `allow_pickle=True`, otherwise `ValueError: Object arrays cannot be loaded when allow_pickle=False`. 
The other option to avoid [numpy pickle security issue](https://numpy.org/devdocs/reference/generated/numpy.load.html) would be to set activity as a float in the array (1. for 'walking, 2. for ...).

Just tell me if you prefer to do so.

Reply:
There are also [structured arrays](https://numpy.org/doc/stable/user/basics.rec.html) to avoid `object` when you want to mix types in an array.
I think I'd rather avoid pickle since we ended up here because we were trying to avoid it (a bit circular...), and avoid having an array of `object`. So either a structured array, or what you were suggesting (suitably documented in both cases). This isn't a very strong opinion though, and I haven't thought about it enough to have a preference between the 2 options.
The last option is to store each array separately, and have 'all' return a list of 4 arrays :stuck_out_tongue_winking_eye:

Reply:
I tried a structured array on [8388b14](https://github.com/GUDHI/gudhi-devel/pull/690/commits/8388b14770f301fb559e846eb51c514e8d2e7482) with the help of https://github.com/GUDHI/gudhi-data/pull/9